django/core/cache/backends/memcached.py

```diff
@@ -213,3 +213,55 @@ def _modify_servers(servers):
                 yield server[5:]
             else:
                 yield server
+
+
+class PyMemcacheCache(BaseMemcachedCache):
+    """An implementation of a cache binding using pymemcache."""
+
+    def __init__(self, server, params):
+        import pymemcache.client.hash
+        import pymemcache.serde
+        super().__init__(server, params, library=pymemcache, value_not_found_exception=KeyError)
+        self._class = type('PyMemcacheClient', (PyMemcacheCompatMixin, self._lib.client.hash.HashClient), {})
+        self._options = {
+            'allow_unicode_keys': True,
+            'default_noreply': False,
+            'serde': pymemcache.serde.pickle_serde,
```

Comment:
We set these to make the behaviour consistent with the other backends and to allow the test suite to pass successfully. Currently `django-pymemcache` only sets the soft-deprecated `serializer` and `deserializer` options which have been superseded by `serde`.

Note that `pymemcache.serde.pickle_serde` uses `pickle.HIGHEST_PROTOCOL` by default which is consistent with the configuration of the `python-memcached` backend.

Reply:
Why set default_noreply to False?

Reply:
For consistency with other memcache client libraries and other cache backends. I wanted this new backend to be a drop in replacement. You can override this if you need to by configuring `OPTIONS`.